src/main/kotlin/com/exactpro/th2/common/message/MessageFilterUtils.kt

```diff
@@ -76,6 +82,28 @@ fun RootMessageFilter.toTreeTable(): TreeTable = TreeTableBuilder().apply {
     row("comparison-settings", comparisonSettings.toTreeTableEntry())
 }.build()
 
+fun RootMessageFilter.toReadableBodyCollection(additionalMetadata: Map<String, String>? = null): Collection<IBodyData> =
+    listOf(
+        TreeTableBuilder().apply {
+            name("Filter")
+            row("message-filter", messageFilter.toTreeTableEntry())
+            row("metadata-filter", metadataFilter.toTreeTableEntry())
+        }.build(),
+        TreeTableBuilder().apply {
+            name("Settings")
+            row("comparison-settings", comparisonSettings.toTreeTableEntry())
+        }.build(),
+        MessageBuilder().apply {
+            text("Metadata")
+        }.build(),
+        TableBuilder<MetadataField>().apply {
+            row(MetadataField("message-type", messageType))
+            additionalMetadata?.forEach {
+                row(MetadataField(it.key, it.value))
+            }
+        }.build()
```

Comment:
Can we use a tree table here as well? There won't be any difference in the result view except we will be able to set the header

Reply:
Also, I think the information about the message type should go first (before filter and its settings)